### PR TITLE
Arch_R and Ipc_R

### PR DESCRIPTION
**Pull request by mktnk3**
This removes some sorries from Arch_R and Ipc_R. Just so we can see a little decrease in the sorry count... :)

Signed-off-by: Miki Tanaka <miki.tanaka@data61.csiro.au>

### CODE REVIEW

File: proof/refine/ARM/Arch_R.thy [hunk 1]
```diff
@@ -259,14 +259,14 @@ lemma pac_corres:
...
        subgoal premises prems using prems by (clarsimp simp:invs_def valid_state_def)+
-  sorry (*
      apply (clarsimp simp:cte_wp_at_caps_of_state)
     apply (drule detype_locale.non_null_present)
      apply (fastforce simp:cte_wp_at_caps_of_state)
     apply simp
    apply (frule_tac ptr = "(aa,ba)" in detype_invariants [rotated 3])
         apply fastforce
        apply simp
+            apply (clarsimp simp: schact_is_rct)
```
Comment (mbrcknl) [inline proof/refine/ARM/Arch_R.thy:269]: Indentation?

File: proof/refine/ARM/Arch_R.thy [hunk 2]
```diff
@@ -335,7 +333,8 @@ lemma pac_corres:
     apply (clarsimp simp: invs'_def valid_state'_def if_unsafe_then_cap'_def cte_wp_at_ctes_of)
    apply fastforce
   apply simp
-  done *)
+  apply simp
+  done
```
Comment (mbrcknl) [inline proof/refine/ARM/Arch_R.thy:337]: Possibly something has gone wrong with indentation here.

**Review by mbrcknl — APPROVED** [review]
Looks good!

File: proof/refine/ARM/Ipc_R.thy [hunk 3]
```diff
@@ -1864,20 +1882,30 @@ lemma cancelAllSignals_weak_sch_act_wf[wp]:
...
+  done
 
 crunches replyUnlink
   for nosch[wp]: "\<lambda>s. P (ksSchedulerAction s)"
   (simp: crunch_simps wp: crunch_wps)
 
-lemma replyUnlink_weak_sch_act_wf[wp]:
-  "replyUnlink r t \<lbrace> \<lambda>s. weak_sch_act_wf (ksSchedulerAction s) s \<rbrace>"
+(* see replyUnlink_weak_sch_act_wf in IpcCancel_R *)
+lemma replyUnlink_weak_sch_act_wf:
```
Comment (michaelmcinerney) [inline proof/refine/ARM/Ipc_R.thy:1898]: Are you sure this one is provable? `replyUnlink` will set the thread state of `p` to `Inactive`

**Review by michaelmcinerney — APPROVED** [review]

File: proof/refine/ARM/Ipc_R.thy (hunk 3 above)
Comment (mktnk3) [reply proof/refine/ARM/Ipc_R.thy:1898]: I don’t think so. I actually tried removing this sorry but it requires a bit more work than I could afford for the purpose of this PR to sort out the dependency. So, yes, this needs to be sorted. I will update the comment to be more specific.

Comment (VictorPhan) [reply proof/refine/ARM/Ipc_R.thy:1898]: I've added a similar lemma in my recent PR merge, but it had an additional `sch_act_not` in the precondition of the hoare triple.

**Review by VictorPhan — APPROVED** [review]

**Review by corlewis — APPROVED** [review]
Nice, it's good to see so many sorries go away!

**Pull request merged**